cmd/list.go

```diff
@@ -82,102 +80,93 @@ func (lr *listResult) print() {
 }
 
 func showComponentList(env *meta.Environment, onlyInstalled bool, refresh bool) (*listResult, error) {
-	if refresh || env.Profile().Manifest() == nil {
-		manifest, err := env.Repository().Manifest()
-		if err != nil {
-			return nil, err
-		}
-		err = env.Profile().SaveManifest(manifest)
-		if err != nil {
-			return nil, err
-		}
+	index, err := env.V1Repository().FetchIndexManifest()
+	if err != nil {
+		return nil, err
 	}
 
 	installed, err := env.Profile().InstalledComponents()
 	if err != nil {
 		return nil, err
 	}
-	manifest := env.Profile().Manifest()
+
 	var cmpTable [][]string
-	cmpTable = append(cmpTable, []string{"Name", "Installed", "Platforms", "Description"})
+	cmpTable = append(cmpTable, []string{"Name", "Owner", "Installed", "Platforms", "Description"})
 
 	localComponents := set.NewStringSet(installed...)
-	for _, comp := range manifest.Components {
-		if comp.Hide {
-			continue
-		}
-		if onlyInstalled && !localComponents.Exist(comp.Name) {
+	for name, comp := range index.Components {
+		if onlyInstalled && !localComponents.Exist(name) {
 			continue
 		}
 		installStatus := ""
-		if localComponents.Exist(comp.Name) {
-			versions, err := env.Profile().InstalledVersions(comp.Name)
+		if localComponents.Exist(name) {
```

Comment:
The 'name' here is actually the id of the component, but after seeing this PR I think it is a mistake to have both name and id for components and we should remove name (I can do that if you agree, it doesn't have to be done in this PR).

Reply:
#267